### DISCUSSION TITLE
Implementing SSR in Vue 3 with Ruby on Rails Backend

### DISCUSSION THREAD

**Discussion by acidball54** [vikejs/vike#1890] [Discussion] [answered]
Hello Community, 

I'm currently working on integrating SSR into my Vue 3 app, which is being served from a Ruby on Rails backend. Instead of using a standalone `index.html` file, my app is rendered using `application.html.erb` from Rails. Most of the SSR guides I found are for standalone Vue apps, and I couldn't find specific documentation for a Rails-integrated setup.

Has anyone tried this kind of setup, or are there any best practices/documentation for this scenario? I'd appreciate any guidance or suggestions for handling SSR when Rails serves the initial HTML template.

Thank you

**Comment by acidball54** [comment]
anyone?

**Comment by brillout** [comment]
It isn't clear to me what you want. SSR and its HTML can only be generated by Vue. So I wonder why you want to have the HTML to be generated by RoR while using SSR at the same time. What you want seems to be inherently contradictory.

See also: https://vike.dev/ruby-on-rails

**Reply by brillout** [reply, marked as answer]
New docs about it: https://vike.dev/servers#non-javascript-backend
